<launch> 

	<!-- Name of the YAML file containing the marker positions -->
	<arg name="YAML_file_name" default="zumy_marker_positions"/>

	<!-- File containing the the marker positions in the trackable's frame of reference -->
	<arg name="marker_positions_file" default="$(find monocular_pose_estimator)/marker_positions/$(arg YAML_file_name).yaml"/> 

	<group ns="robot0" >
		<node name="estimator0" pkg="monocular_pose_estimator" type="monocular_pose_estimator" respawn="false" output="screen"> 
			<rosparam command="load" file="$(arg marker_positions_file)"/>
			<param name= "threshold_value" value = "200" />
			<param name= "gaussian_sigma" value = "0.6" />
			<param name= "min_blob_area" value = "9" />
			<param name= "max_blob_area" value = "3000" />
			<param name= "max_width_height_distortion" value = "0.75" />
			<param name= "max_circular_distortion" value = "0.75" />
			<param name= "back_projection_pixel_tolerance" value = "5" />
			<param name= "nearest_neighbour_pixel_tolerance" value = "7" />
			<param name= "certainty_threshold" value = "0.25" />
			<param name= "valid_correspondence_threshold" value = "0.25" />
      <param name= "roi_border_thickness" value="20" />
      <remap from="/camera/camera_info" to="/usb_cam/camera_info" />
      <remap from="/camera/image_raw" to="/robot_00/camera_0/image_transport" />
		</node>
	
		<node name="robot0" pkg="image_view" type="image_view" args="image:=image_with_detections" />
	</group>

</launch>
